docs/reference/api/json-rpc.md

```diff
@@ -129,6 +129,80 @@ curl -X POST --data '{"jsonrpc":"2.0","method":"eth_signTransaction","params":[{
 
 <!--/tabs-->
 
+## `eth_signTypedData`
+
+`eth_signTypedData` is the same as [`eth_sign`](#ethsign) except a typed data structure is specified as the object to be signed instead of a data string. 
+
+Calculates an Ethereum specific signature using
+`sign(keccak256("\x19Ethereum Signed Message:\n" + len(message) + message)))."`
+
+Adds a prefix to the message that makes the calculated signature recognizable as an Ethereum
+specific signature.
+This prevents malicious dapps from signing arbitrary data (for example, a transaction) and using the
+signature to impersonate the victim.
+
+The JSON schema for the typed data structure to sign is: 
+
+```json
+{
+  type: 'object',
+  properties: {
+    types: {
+      type: 'object',
+      properties: {
+        EIP712Domain: {type: 'array'},
+      },
+      additionalProperties: {
+        type: 'array',
+        items: {
+          type: 'object',
+          properties: {
+            name: {type: 'string'},
+            type: {type: 'string'}
+          },
+          required: ['name', 'type']
+        }
+      },
+      required: ['EIP712Domain']
+    },
+    primaryType: {type: 'string'},
+    domain: {type: 'object'},
+    message: {type: 'object'}
+  },
+  required: ['types', 'primaryType', 'domain', 'message']
+}
+```
+
+### Parameters
+
+`DATA` - 20-byte account address
+
+`TYPEDDATA` - Typed data structure to sign
+
+### Returns
+
+`DATA` - Signature
+
+<!--tabs-->
+
+# curl HTTP request
+
+```bash
+curl -X POST --data '{"jsonrpc":"2.0","method":"eth_signTypedData","params":["0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826", {"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Person":[{"name":"name","type":"string"},{"name":"wallet","type":"address"}],"Mail":[{"name":"from","type":"Person"},{"name":"to","type":"Person"},{"name":"contents","type":"string"}]},"primaryType":"Mail","domain":{"name":"Ether Mail","version":"1","chainId":1,"verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC"},"message":{"from":{"name":"Cow","wallet":"0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"},"to":{"name":"Bob","wallet":"0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"},"contents":"Hello, Bob!"}}],"id":1}'
```

Comment:
@Gabriel-Trintinalia - I added this example before submitting for review (forgot to mention in the comments) -> will this work?  Or did you see this and think another example is required?

Reply:
tagging @gfukushima :)

Reply:
@Gabriel-Trintinalia - sorry!!

Reply:
I guess I missed the example then!